pilot/pkg/features/experimental.go

```diff
@@ -197,4 +197,7 @@ var (
 	EnableLocalityWeightedLbConfig = env.Register("ENABLE_LOCALITY_WEIGHTED_LB_CONFIG", false,
 		"If enabled, always set LocalityWeightedLbConfig for a cluster, "+
 			" otherwise only apply it when locality lb is specified by DestinationRule for a service").Get()
+
+	UnifiedSidecarScoping = env.Register("PILOT_UNIFIED_SIDECAR_SCOPE", true,
```

Comment:
my pitch: we want to move to this model because its more correct and fixes real bugs for real users. It also reduces tech debt, though that is a minor impact. We have a flag since its a very high risk change, with known intentional behavioral changes (though these are quite niche).

Making it off by default doesn't further either of these goals. Typically I think off-by-default feature flags are only useful for changes that are experimental new features. This isn't really in that category, its instead a change we just want an escape hatch to be able to turn it off if we messed up and caused a regression

Reply:
> But this may introduce regressions

I don't disagree but its a bit vague, so I am not sure how to address this comment.

Do you mean there are unknown-unknowns that we may accidentally break things unexpected? If so, that is the point of the feature flag -- if a user reports a bug, we can have them set it to false easily and work on a fix. I am not sure there is much else we can do to control unknown-unknowns.

Do you mean there is a specific part of this change you feel is going to break users? If so, can you give some more details about what part so I can work to address it?